packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -238,14 +246,50 @@ String generatePluralMethod(Message message, AppResourceBundle bundle) {
     .replaceAll('@(none)\n', '');
 }
 
+bool needsCurlyBracketStringInterpolation(String messageString, String placeholder, bool isPlural) {
+  final int placeholderIndex = messageString.indexOf(placeholder);
+  // This means that there is no need for a placeholder, since one was not found
+  // in the message
+  if (placeholderIndex == -1) {
```

Comment:
I don't understand why `return messageString.indexOf('\{$placeholder\}') != -1` is insufficient, or perhaps something similar that allows for whitespace around the placeholder?

Reply:
Discussed offline -- Basically, it's not just whitespace that needs to be checked for -- There is the cases where the placeholder is at the beginning of the string, in the middle of the string, and at the end of the string. The checks are slightly different for each case.

Also, it's not just whitespace -- As long as the characters are not alphabets and underscores (a-z, A-Z, and _), then it should be okay to not use curly braces, since the Dart code would treat everything else as part of the string and not part of the variable name.